### PR TITLE
My implementation and clean code in the test, I think...

### PR DESCRIPTION
**Pull request by josegarm**

### CODE REVIEW

File: java/test/problems/problem2/NumberOfMarksCost.java [hunk 1]
```diff
@@ -36,25 +38,28 @@ public static void main(String args[]) {
...
+                boolean correct = true;
+                for(int r = 0; r < REPETITIONS; r++) {
+                    //The mark should be random and not predefined
+                    int mark = random.nextInt(MAX_ARRAY-1);
+                    vector = generateIncreasingRandoms(t, MAX_ARRAY);
+                    t1 = System.nanoTime();
+                    int n = ex.numberOfMarks(vector, mark);
+                    t2 = System.nanoTime();
+                    time += t2 - t1;
+                    if(!(vector[vector.length-n-1]<=mark && vector[vector.length-n]>mark)){
```
Comment (tonilopezmr) [inline java/test/problems/problem2/NumberOfMarksCost.java]: This is a stupid if, isn't it?

boolean correct = (vector[vector.length-n-1]<=mark && vector[vector.length-n]>mark)

Comment (josegarm) [inline java/test/problems/problem2/NumberOfMarksCost.java]: hahaha, is possible. You can change it, but this condition work correctly.

Comment (tonilopezmr) [inline java/test/problems/problem2/NumberOfMarksCost.java]: xD you are a fucker, it's not about if work correctly or not, it's about good code o 💩 code.

File: java/test/problems/problem2/NumberOfMarksCost.java [hunk 2]
```diff
@@ -36,25 +38,26 @@ public static void main(String args[]) {
...
+                boolean correct = true;
+                for(int r = 0; r < REPETITIONS; r++) {
+                    //The mark should be random and not predefined
+                    int mark = random.nextInt(MAX_ARRAY-1);
+                    vector = generateIncreasingRandoms(t, MAX_ARRAY);
+                    t1 = System.nanoTime();
+                    int n = ex.numberOfMarks(vector, mark);
+                    t2 = System.nanoTime();
+                    time += t2 - t1;
+                    correct = correct && (vector[vector.length-n-1]>mark || vector[vector.length-n]<=mark);
```
Comment (josegarm) [inline java/test/problems/problem2/NumberOfMarksCost.java]: Boolean correct has been corrected.

**Comment by tonilopezmr** [comment]
Well done @josegarm !!! 

Issue #26 resolved

!()[http://www.shortscore.net/wp-content/uploads/2014/04/Maennerbeule__48_.gif]

**Pull request merged**